beamer/tests/check/test_invalidations.py

```diff
@@ -0,0 +1,63 @@
+import itertools
+import json
+from itertools import repeat
+from pathlib import Path
+
+import ape
+import apischema
+import pytest
+
+import beamer.check.commands
+from beamer.check.commands import Invalidation
+from beamer.tests.util import deploy, get_repo_root, run_command, write_keystore_file
+
+
+@pytest.mark.parametrize("count", (1, 3))
+def test_initiate_l1_invalidations(tmp_path, deployer, count):
+    password = "test"
+    keystore_file = tmp_path / f"{deployer.address}.json"
+    write_keystore_file(keystore_file, deployer.private_key, password)
+
+    rpc_file, artifact = deploy(deployer, tmp_path)
+
+    root = get_repo_root()
+    output = tmp_path / "invalidations.json"
+    chain_id = ape.chain.chain_id
+    # We run the command with 2 proof targets: current chain and a dummy chain.
+    proof_targets = chain_id, 98765
+    run_command(
+        beamer.check.commands.initiate_l1_invalidations,
+        "--keystore-file",
+        keystore_file,
+        "--password",
+        password,
+        "--rpc-file",
+        rpc_file,
+        "--abi-dir",
+        f"{root}/contracts/.build/",
+        "--artifacts-dir",
+        Path(artifact).parent,
+        "--output",
+        output,
+        "--count",
+        count,
+        str(chain_id),
+        *map(str, proof_targets),
+    )
+    with output.open("rt") as f:
+        data = json.load(f)
+
+    assert len(data) == count * len(proof_targets)
+    w3 = ape.chain.provider.web3
+    invalidations = apischema.deserialize(list[Invalidation], data)
+    targets = itertools.chain.from_iterable(repeat(x, count) for x in proof_targets)
+    for invalidation, proof_target in zip(invalidations, targets):
+        assert invalidation.proof_source == chain_id
+        assert invalidation.proof_target == proof_target
+
+        receipt = w3.eth.get_transaction_receipt(invalidation.txhash)
+        fill_manager = ape.project.FillManager.at(receipt.to)
+        event = fill_manager.FillInvalidated.from_receipt(receipt)[0]
+        assert event.event_arguments == dict(
+            requestId=invalidation.request_id, fillId=invalidation.fill_id
+        )
```

Comment:
Suggestion: A test which continues on an existing invalidation file would be nice. 

option a) run the command again with an increased count
option b) run the command again with one more dummy chain

Reply:
Done. Thanks for making me do the tests.

We now have nice coverage:
![image](https://github.com/beamer-bridge/beamer/assets/1312091/05b253ea-577f-4b63-89ba-68dc63f7b047)

(The uncovered part is mostly OP-prove-related code.)